### PR TITLE
Run `go generate` and `go vet` on all packages

### LINKED ISSUE #30512: To clone the project locally, execute 'make build' and get the following error
### Description

To clone the project locally, execute 'make build' and get the following error:

`Running go generate...
no required module provides package code.gitea.i; to add it:
        go get code.gitea.i
make: *** [makefile:782: generate-go] Error 1`

image

### Gitea Version

master

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

My running environment: 

win10、Git Bash、make 4.2、go 1.22

### Database

None

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/30512

I think this does mean those tools would run on a potential `vendor` directory, but I'm not sure we really support vendoring of dependencies anymore.

`release` has a `vendor` prerequisite so likely the source tarballs contain vendor files?

### CODE REVIEW

**Comment by silverwind** [comment]
On the other hand, I'm pretty sure golang tools would ignore files in the well-known `vendor` directory itself, at least for `go vet`.

File: tests/integration/repo_mergecommit_revert_test.go [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
```
Comment (silverwind) [inline tests/integration/repo_mergecommit_revert_test.go:2]:
```suggestion
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.
// SPDX-License-Identifier: MIT
```

File: tests/integration/repo_mergecommit_revert_test.go [hunk 2]
```diff
@@ -1,3 +1,8 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
```
Comment (delvh) [inline tests/integration/repo_mergecommit_revert_test.go]: I think we can remove these two lines now in all the 5 test files

**Review by delvh — APPROVED** [review]

File: tests/integration/api_comment_attachment_test.go [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2021 The Gitea Authors. All rights reserved.
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
```
Comment (silverwind) [inline tests/integration/api_comment_attachment_test.go]:
```suggestion
```

File: tests/integration/api_issue_attachment_test.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2021 The Gitea Authors. All rights reserved.
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
```
Comment (silverwind) [inline tests/integration/api_issue_attachment_test.go]:
```suggestion
```

File: tests/integration/api_packages_cargo_test.go [hunk 5]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2021 The Gitea Authors. All rights reserved.
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
```
Comment (silverwind) [inline tests/integration/api_packages_cargo_test.go]:
```suggestion
```

File: tests/integration/markup_external_test.go [hunk 6]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2022 The Gitea Authors. All rights reserved.
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
```
Comment (silverwind) [inline tests/integration/markup_external_test.go]:
```suggestion
```

File: tests/integration/repo_mergecommit_revert_test.go (hunk 2 above)
Comment (silverwind) [inline tests/integration/repo_mergecommit_revert_test.go]:
```suggestion
```

Comment (silverwind) [reply tests/integration/repo_mergecommit_revert_test.go]: done

**Review by lunny — APPROVED** [review]

**Pull request merged**